### PR TITLE
DEVCVL-267 - Auction Creation

### PR DESCRIPTION
**Pull request by kyawthetkt**

### CODE REVIEW

File: src/components/product/bid/Bid.js [hunk 1]
```diff
@@ -110,7 +104,7 @@ const Bid = ({
                     />
                 </Row>
                 <Row></Row>
-                {highestBidder === username && (
+                {username && highestBidder === username && (
```
Comment (jibrail-idris) [inline src/components/product/bid/Bid.js:107]: This equality match can be problematic. I have seen addresses having their cases randomly assigned.
Perhaps to be safer, apply lowercase to both `highestBidder` and `username`

File: src/contracts/hooks/useApproval.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    const [error, setError] = useState(null);
+    const { Moralis } = useMoralis();
+
+    const init = async (contractAddress, tokenId, lazyMinted = false) => {
+        try {
+            await Moralis.enableWeb3();
+
+            const transaction = await Moralis.executeFunction({
+                contractAddress: contractAddress,
+                functionName: "approve",
```
Comment (jibrail-idris) [inline src/contracts/hooks/useApproval.js:21]: Why use lazy minted flag to determine the ERC721 type? and also we don't have lazy minting yet.

File: src/components/product/bid/Bid.js (hunk 1 above)
Comment (kyawthetkt) [reply src/components/product/bid/Bid.js:107]: @jibrail-idris Thanks for such a big spot.Will fix it!

File: src/contracts/hooks/useApproval.js (hunk 2 above)
Comment (kyawthetkt) [reply src/contracts/hooks/useApproval.js:21]: will remove  `lazyMinted` flag as even though we have a lazy minting, **ERC721** no longer needed. .

File: src/App.js [hunk 3]
```diff
@@ -100,6 +101,10 @@ const App = () => (
                                 path="asset/:assetId/edit"
                                 element={<CreateEditAsset />}
                             />
+                            <Route
+                                path="/collection/:collectionName/asset/:assetId/auction/create"
```
[the remark this answers is not in the document]
Comment (kyawthetkt) [reply src/App.js:105]: @shakil-adv May I know your suggestion for this route?

[the remark this answers is not in the document]
Comment (kyawthetkt) [reply src/App.js:105]: @shakil-adv May I know suggestion for this route?

[the remark this answers is not in the document]
Comment (shakil-adv) [reply src/App.js:105]: Okay then let's let it be as it is. We can combine routes later

**Review by jibrail-idris — APPROVED** [review]

**Review by shakil-adv — APPROVED** [review]

**Review by pracheebapate — APPROVED** [review]

**Pull request merged**